environ/process/asset_pricing/double_sorting.py

```diff
@@ -264,111 +227,89 @@ def _eval_port(
     # calculate the bottom minus top
     df_ret["top_minus_bottom"] = df_ret["top"] - df_ret["bottom"]
 
-    plot_dict = {
-        "regular": [
-            "top",
-            "bottom",
-            "top_minus_bottom",
-        ],
-        "stable": [
-            "True_top",
-            "True_bottom",
-        ],
-        "nonstable": [
-            "False_top",
-            "False_bottom",
-        ],
-    }
+    plot_lst = ["top", "bottom", "top_minus_bottom"]
 
     # calculate the cumulative return
-    for _, lst in plot_dict.items():
-        for col_ret in lst:
-            df_ret[col_ret + "_cum"] = (df_ret[col_ret] + 1).cumprod()
+    for col_ret in plot_lst:
+        df_ret[col_ret + "_cum"] = (df_ret[col_ret] + 1).cumprod()
 
     # three subplots for cum ret sharing x axis
-    _, (ax_ret, ax_ret_stable, ax_ret_nonstable) = plt.subplots(
-        nrows=3, ncols=1, sharex=True, figsize=(12, 7)
-    )
+    (
+        _,
+        ax_ret,
+    ) = plt.subplots(figsize=(12, 7))
 
-    for col in plot_dict["regular"]:
+    for col in plot_lst:
         ax_ret.plot(df_ret["freq"], df_ret[col + "_cum"], label=col)
 
-    for col in plot_dict["stable"]:
-        ax_ret_stable.plot(df_ret["freq"], df_ret[col + "_cum"], label=col)
-
-    for col in plot_dict["nonstable"]:
-        ax_ret_nonstable.plot(df_ret["freq"], df_ret[col + "_cum"], label=col)
-
-    for plt_subplot in [ax_ret, ax_ret_stable, ax_ret_nonstable]:
-        # plot boom bust cycles
-        plot_boom_bust(plt_subplot, boom_bust=BOOM_BUST)
-        plt_subplot.set_xlim(df_panel["Date"].min(), df_panel["Date"].max())
+    # plot boom bust cycles
+    plot_boom_bust(ax_ret, boom_bust=BOOM_BUST)
+    ax_ret.set_xlim(df_panel["Date"].min(), df_panel["Date"].max())
 
     # legend
     ax_ret.legend()
-    ax_ret_stable.legend()
-    ax_ret_nonstable.legend()
 
     # tight layout
     plt.tight_layout()
 
     # save the plot to the save path
     plt.savefig(save_path, dpi=300)
 
-    # show the plot
-    plt.show()
-
     # close the plot
     plt.close()
 
 
 def asset_pricing(
+    reg_panel: pd.DataFrame,
     save_path: str,
```

Comment:
again why not `Path | str`?
then we don't have convert path to str which i find unnecessary

Reply:
Fixed.